src/EditorFeatures/Core/EditAndContinue/EditAndContinueLanguageService.cs

```diff
@@ -102,12 +97,18 @@ internal void Disable(Exception e)
             .ReportNonFatalErrorAsync().CompletesAsyncOperation(token);
     }
 
+    private void UpdateApplyChangesDiagnostics(ImmutableArray<DiagnosticData> diagnostics)
+    {
+        sessionState.ApplyChangesDiagnostics = diagnostics;
+        diagnosticRefresher.RequestWorkspaceRefresh();
```

Comment:
Remind me, where we are saving the state that indicates to the pull handler that the diagnostics have changed (and needs to be recomputed)?  If we only send the refresh notification (without any doc or project change), the handler will return nothing changed
https://github.com/dotnet/roslyn/blob/75a79bbfba8cd1958655d1456299fda5c5fb7c3a/src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs#L182-L188

Reply:
RequestWorkspaceRefresh increments global version before triggering the refresh notification.